mm2src/lp_swap/taker_swap.rs

```diff
@@ -1684,15 +1694,19 @@ pub fn max_taker_vol_from_available(
     let fee_threshold = dex_fee_threshold(min_tx_amount.clone());
     let dex_fee_rate = dex_fee_rate(base, rel);
     let threshold_coef = &(&MmNumber::from(1) + &dex_fee_rate) / &dex_fee_rate;
-    let mut max_vol = if available > &fee_threshold * &threshold_coef {
+    let max_vol = if available > &fee_threshold * &threshold_coef {
         available / (MmNumber::from(1) + dex_fee_rate)
     } else {
         available - fee_threshold
     };
 
     if &max_vol <= min_tx_amount {
-        warn!("max_vol {} <= min_tx_amount {}", max_vol, min_tx_amount);
-        max_vol = 0.into();
+        let err = ERRL!(
+            "Max taker volume {:?} less than minimum transaction amount {:?}",
+            max_vol.to_fraction(),
+            min_tx_amount.to_fraction()
+        );
+        return Err(CheckBalanceError::NotSufficientBalance(err));
```

Comment:
Does it mean that `trade_preimage` RPC will return an error in the `max` case when the balance is not sufficient?

Reply:
Yes, it is. I think, `volume = 0` makes sense, but `volume = 0, base_coin_fee.amount = 0, rel_coin_fee.amount = 0` doesn't

Reply:
Yeah, I agree on this, thanks!